tests/test_core.py

```diff
@@ -783,6 +783,21 @@ def __init__(self):
         imageio.imwrite("foo.bmp", "asd")
 
 
+def test_imwrite_symbol_name():
+    # this is a regression test for https://github.com/imageio/imageio/issues/674
+    if sys.platform == "linux":
+        name = """#!~:@$%^&`-+{};',.()? []_=.jpg"""
+    elif sys.platform == "win32":
+        name = """#!~@$%^&`-+{};',.() []_=.jpg"""
+    elif sys.platform == "darwin":
+        name = """#!~@$%^&`-+{};',.()? []_=.jpg"""
+    else:
+        pytest.skip("Unknown OS.")
+    tmp_request = imageio.core.Request(name, "r")
```

Comment:
```suggestion
    tmp_request = imageio.core.Request(name, "w")
```

The CI fails when creating the read request, because it checks if the file exists upon construction (It should probably do that upon first read instead). Writing will not perform this check, and we technically don't need it here, so preparing to write is probably the better thing to do.